storage/src/key_value/agent/block.rs

```diff
@@ -223,34 +227,49 @@ impl<S: KeyValueStorage + 'static> Agent<S> {
         })
     }
 
-    pub(super) fn get_block_locator_hashes(&mut self) -> Result<Vec<Digest>> {
+    pub(super) fn get_block_locator_hashes(
+        &mut self,
+        points_of_interest: Vec<Digest>,
+        oldest_fork_threshold: usize,
+    ) -> Result<Vec<Digest>> {
         let canon = self.canon()?;
+        let target_height = canon.block_height as u32;
 
         // The number of locator hashes left to obtain; accounts for the genesis block.
-        let mut num_locator_hashes = std::cmp::min(crate::NUM_LOCATOR_HASHES - 1, canon.block_height as u32);
+        let mut num_locator_hashes = std::cmp::min(crate::NUM_LOCATOR_HASHES - 1, target_height);
 
         // The output list of block locator hashes.
-        let mut block_locator_hashes = Vec::with_capacity(num_locator_hashes as usize);
+        let mut block_locator_hashes = Vec::with_capacity(num_locator_hashes as usize + points_of_interest.len());
+
+        for hash in points_of_interest {
+            trace!("block locator hash -- interesting: block# none: {}", hash);
+            block_locator_hashes.push(hash);
```

Comment:
should the fork hashes be provided before the top hashes? those will probably constitute the happy path

Reply:
we don't want to be sidetracked by a fork -- it ultimately shouldn't matter though.